martin-observability-utils/src/lib.rs

```diff
@@ -0,0 +1,292 @@
+#![doc = include_str!("../README.md")]
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+use tracing_subscriber::EnvFilter;
+
+#[derive(Default)]
+pub struct MartinObservability {
+    filter: EnvFilter,
+    log_format: LogFormatOptions,
+}
+impl MartinObservability {
+    /// transform [`log`](https://docs.rs/log) records into [`tracing`](https://docs.rs/tracing) [`Event`](tracing::Event)s.
+    ///
+    /// # Panics
+    /// This function will panic if the global `log`-logger cannot be set.
+    /// This only happens if the global `log`-logger has already been set.
+    #[must_use]
+    pub fn with_initialised_log_tracing(self) -> Self {
+        tracing_log::LogTracer::builder()
+            .with_interest_cache(tracing_log::InterestCacheConfig::default())
+            .init()
+            .expect("the global logger to only be set once");
+        self
+    }
+    /// Set the global subscriber for the application.
+    ///
+    /// # Panics
+    /// This function will panic if the global subscriber cannot be set.
+    /// This only happens if the global subscriber has already been set.
+    pub fn set_global_subscriber(self) {
+        use tracing::subscriber::set_global_default;
+        use tracing_subscriber::fmt::Layer;
+        use tracing_subscriber::prelude::*;
+        let registry = tracing_subscriber::registry().with(self.filter);
+        match self.log_format {
+            LogFormatOptions::Full => set_global_default(registry.with(Layer::default())),
+            LogFormatOptions::Compact => {
+                set_global_default(registry.with(Layer::default().compact()))
+            }
+            LogFormatOptions::Pretty => {
+                set_global_default(registry.with(Layer::default().pretty()))
+            }
+            LogFormatOptions::Json => set_global_default(registry.with(Layer::default().json())),
+        }
+        .expect("since martin has not set the global_default, no global default is set");
+    }
+}
+impl From<(EnvFilter, LogFormatOptions)> for MartinObservability {
+    fn from((filter, log_format): (EnvFilter, LogFormatOptions)) -> Self {
+        Self { filter, log_format }
+    }
+}
+
+#[derive(
+    PartialEq,
+    Eq,
+    Clone,
+    Copy,
+    Default,
+    Debug,
+    clap::ValueEnum,
+    serde::Serialize,
+    serde::Deserialize,
+)]
+pub enum LogFormatOptions {
```

Comment:
Currently the `bare` formatting option (added to support mbtiles as before) looks like this:

![image](https://github.com/user-attachments/assets/7cb9f5a7-9bae-4507-b47e-07e21ccf907b)


We could remove the target via adding this to the layer
`.map_event_format(|e| e.with_target(false))`

=> should mbtiles remain "as-is" or gain targets?